testsuite/MDAnalysisTests/analysis/test_align.py

```diff
@@ -151,6 +151,19 @@ def test_toggle_atom_nomatch_mismatch_atoms(self, universe, reference):
         with pytest.raises(SelectionError):
             align.alignto(u, ref, select='all', match_atoms=False)
 
+    def test_subselection_alignto(self, universe, reference):
+        sub_string = 'resname ALA and name CA'
+        rmsd = align.alignto(universe, reference, subselection=sub_string)
```

Comment:
The core devs may prefer that you [parametrize](https://docs.pytest.org/en/latest/parametrize.html) this test as well to reduce duplication and have a separate test issued for each case in the test suite. Probably not critical, but I think it is fairly prevalent in MDAnalysis.

I think you can even do that for the `TypeError` case with an extra mark.